src/_gettsim/shared.py

```diff
@@ -39,6 +42,127 @@ def inner(func):
+    change_name
+        The name that should be used as the key for the function in the DAG.
+        If omitted, we use the name of the function as defined.
+
+    Returns
+    -------
+        The function with attributes __info__["dates_active_start"],
+        __info__["dates_active_end"], and __info__["dates_active_dag_key"].
+    """
+
+    _validate_dashed_iso_date(start)
+    _validate_dashed_iso_date(end)
+
+    start_date = date.fromisoformat(start)
+    end_date = date.fromisoformat(end)
+
+    _validate_date_range(start_date, end_date)
+
+    def inner(func: Callable) -> Callable:
+        dag_key = change_name if change_name else func.__name__
+
+        _check_for_conflicts_in_time_dependent_functions(
+            dag_key, func.__name__, start_date, end_date
+        )
+
+        # Remember data from decorator
+        if not hasattr(func, "__info__"):
+            func.__info__ = {}
+        func.__info__["dates_active_start"] = start_date
+        func.__info__["dates_active_end"] = end_date
+        func.__info__["dates_active_dag_key"] = dag_key
+
+        # Register time-dependent function
+        if dag_key not in TIME_DEPENDENT_FUNCTIONS:
+            TIME_DEPENDENT_FUNCTIONS[dag_key] = []
+        TIME_DEPENDENT_FUNCTIONS[dag_key].append(func)
+
+        return func
+
+    return inner
+
+
+_dashed_iso_date = re.compile(r"\d{4}-\d{2}-\d{2}")
+
+
+def _validate_dashed_iso_date(date_str: str):
+    if not _dashed_iso_date.match(date_str):
+        raise ValueError(  # noqa: TRY003
+            f"Date {date_str} does not match the format YYYY-MM-DD."
+        )
+
+
+def _validate_date_range(start: date, end: date):
+    if start > end:
+        raise ValueError(  # noqa: TRY003
+            f"The start date {start} must be before the end date {end}."
+        )
+
+
+def _check_for_conflicts_in_time_dependent_functions(
+    dag_key: str, function_name: str, start: date, end: date
+):
+    if dag_key not in TIME_DEPENDENT_FUNCTIONS:
+        return
+
+    for f in TIME_DEPENDENT_FUNCTIONS[dag_key]:
+        # A function is not conflicting with itself. We compare names instead of
+        # identities since functions might get wrapped, which would change their
+        # identity but not their name.
+        if f.__name__ != function_name and (
+            start <= f.__info__["dates_active_start"] <= end
+            or f.__info__["dates_active_start"]
+            <= start
+            <= f.__info__["dates_active_end"]
+        ):
+            raise ConflictingTimeDependentFunctionsError(
+                dag_key,
+                function_name,
+                start,
+                end,
+                f.__name__,
+                f.__info__["dates_active_start"],
+                f.__info__["dates_active_end"],
+            )
+
+
+class ConflictingTimeDependentFunctionsError(Exception):
+    """Raised when two time-dependent functions have overlapping time ranges."""
```

Comment:
Ahh, I am basically missing this sentence in the above comment. Then it becomes obvious of what is going on. Could you just repeat it there?